backend/selectgen.ml

```diff
@@ -417,15 +417,16 @@ method is_simple_expr = function
       | Cextcall { effects = No_effects; coeffects = No_coeffects; } ->
         List.for_all self#is_simple_expr args
         (* The following may have side effects *)
-      | Capply _ | Cextcall _ | Calloc | Cstore _ | Craise _ | Cprobe _
-      | Cprobe_is_enabled _ | Copaque -> false
+      | Capply _ | Cextcall _ | Calloc | Cstore _
+      | Craise _ | Ccheckbound
+      | Cprobe _ | Cprobe_is_enabled _ | Copaque -> false
```

Comment:
Not in your patch and safe as it is, but out of
curiosity: is `Cprobe_is_enabled` effectful?

Reply:
Only co-effects. It reads a global variable that can be modified by someone else.

Reply:
Noted, thanks.